File: NWorldPermissions/src/top/nololiyt/worldpermissions/entities/DotDividedStringBuilder.java
package top.nololiyt.worldpermissions.entities;

public class DotDividedStringBuilder
{
    private StringBuilder stringBuilder;
    
    @Override
    public String toString()
    {
        return stringBuilder.toString();
    }
    
    public DotDividedStringBuilder append(String name)
    {
        this.stringBuilder.append('.')
                .append(name);
        return this;
    }
    
    public DotDividedStringBuilder(String root)
    {
        this.stringBuilder = new StringBuilder();
        this.stringBuilder.append(root);
    }
    
    public DotDividedStringBuilder(DotDividedStringBuilder builder)
    {
        this.stringBuilder = new StringBuilder();
        this.stringBuilder.append(builder.stringBuilder.toString());
    }
}

File: NWorldPermissions/src/top/nololiyt/worldpermissions/entities/DotDividedStringBuilder.kt
package top.nololiyt.worldpermissions.entities

class DotDividedStringBuilder {
    private var stringBuilder: StringBuilder? = null

    override fun toString(): String {
        return stringBuilder!!.toString()
    }

    fun append(name: String): DotDividedStringBuilder {
        this.stringBuilder!!.append('.')
                .append(name)
        return this
    }

    constructor(root: String) {
        this.stringBuilder = StringBuilder()
        this.stringBuilder!!.append(root)
    }

    constructor(builder: DotDividedStringBuilder) {
        this.stringBuilder = StringBuilder()
        this.stringBuilder!!.append(builder.stringBuilder!!.toString())
    }
}